### PR TITLE
NN-1834: Add in create user screen

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/Admin/MaintainAuthUsers/containers/AuthUserCreateContainer.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.handleCreate = this.handleCreate.bind(this)
+    this.state = {}
+  }
+
+  async componentDidMount() {
+    const { dispatchLoaded } = this.props
+    dispatchLoaded(true)
+  }
+
+  handleChange(event) {
```
Comment (jsrobertson) [inline src/Admin/MaintainAuthUsers/containers/AuthUserCreateContainer.js]: Not suggesting changing now but we should probably agree on which way we want to bind these when we make a new component.  I tend to write them as arrow functions (class properties) which saves having to bind them in the constructor.

File: backend/controllers/authUserMaintenance.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
   try {
     await apiCall()
   } catch (e) {
-    if (e.response && e.response.status < 500) {
+    if (e.response && e.response.data && e.response.status < 500) {
       res.status(e.response.status)
-      const field = e.response.data.field ? e.response.data.field : defaultField
-      res.json([{ targetName: field, text: e.response.data.error_description }])
+      // eslint-disable-next-line camelcase
+      const { field, error, error_description } = e.response.data
```
Comment (andrewrlee) [inline backend/controllers/authUserMaintenance.js]: meh, could you remove the eslint hint if you use aliasing? it is supposed to ignore it: https://github.com/eslint/eslint/issues/3185

File: src/Admin/MaintainAuthUsers/containers/AuthUserValidation.js [hunk 3]
```diff
@@ -18,4 +18,55 @@ const validateAdd = (role, setError) => {
...
+      targetName: 'email',
+      text: 'Enter an email address in the correct format, like first.last@justice.gov.uk',
+    })
+  }
+  if (!email.match(/[0-9A-Za-z@.'_\-+]*/)) {
+    errors.push({
+      targetName: 'email',
+      text: "Email address can only contain 0-9, a-z, @, ', _ and . characters",
+    })
+  }
```
Comment (Nimphal) [inline src/Admin/MaintainAuthUsers/containers/AuthUserValidation.js]: Do we not allow comments in emails (using +) on purpose?

Comment (petergphillips) [reply src/Admin/MaintainAuthUsers/containers/AuthUserValidation.js]: Well spotted, the regexp allows them but the message says not.  Will update the message

File: backend/controllers/authUserMaintenance.js (hunk 2 above)
Comment (petergphillips) [reply backend/controllers/authUserMaintenance.js]: Good call.  The problem was that I wasn't using aliasing, just destructuring into a variable of the same name.  Will fix.

File: src/Admin/MaintainAuthUsers/containers/AuthUserCreateContainer.js (hunk 1 above)
Comment (petergphillips) [reply src/Admin/MaintainAuthUsers/containers/AuthUserCreateContainer.js]: I think I tried that, but it didn't work.  Will investigate and correct in the next PR.

**Pull request merged**